src/split-criterions/GiniImpurity.jl

```diff
@@ -52,14 +50,14 @@ Args:
 Returns:
     A tuple containing the left and right sets of labels.
   """
-function split_node(data::Matrix{T}, labels::Vector{L}, index, value) where {T, L} 
+function split_node(data::AbstractMatrix, labels::AbstractVector, index, value)
```

Comment:
Maybe add types to index and value as well

Reply:
changed that